session/http_server.go

```diff
@@ -165,9 +165,10 @@ const cookieSessionName = "influxdb-oss-session"
 
 func encodeCookieSession(w http.ResponseWriter, s *influxdb.Session) {
 	c := &http.Cookie{
-		Name:  cookieSessionName,
-		Value: s.Key,
-		Path:  "/api/",
+		Name:    cookieSessionName,
+		Value:   s.Key,
+		Path:    "/api/",
+		Expires: s.ExpiresAt,
```

Comment:
This has the effect of making the cookie persistent (see https://cheatsheetseries.owasp.org/cheatsheets/Session_Management_Cheat_Sheet.html#expire-and-max-age-attributes). Specifically:

"
_If a cookie presents the [Max-Age](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie#Directives) (that has preference over [Expires](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie#Directives)) or Expires attributes, it will be considered a persistent cookie and will be stored on disk by the web browser based until the expiration time._

_Typically, session management capabilities to track users after authentication make use of non-persistent cookies. This forces the session to disappear from the client if the current web browser instance is closed. Therefore, it is highly recommended to use non-persistent cookies for session management purposes, so that the session ID does not remain on the web client cache for long periods of time, from where an attacker can obtain it._
"

While OWASP's recommendation makes sense in general, our implementation stores the actual session server side so there is no sensitive information in the cookie (eg, just the name, session id and path) so I think setting `Expires` is ok if it helps to address a usability issue. There seems to be a slight preference in the community to use `max-age` instead of `Expires`, and indeed it makes conceptual sense to take `--session-length`, convert that to seconds (as per https://go.dev/src/net/http/cookie.go) and shove that into `MaxAge` in the `http.Cookie`. However, `session/service.go:CreateSession()` is already doing the math and giving you `ExpiresAt` for free. `Expires` works and it simplifies our implementation so using it seems fine.

Reply:
> [1] by definition, the session id is sensitive because it gives access to the authenticated login session. Storing it on disk means that a security issue in the browser could expose the contents of the cookie from the disk. The default session length of 1 hour means that cookie will only stick around until it is refreshed, so shorter values mitigate this risk. Of course, a different browser vulnerability could allow getting the cookie for memory or preventing the cookie from expiring, so non-persistent cookies aren't invulnerable to these issues. Also note that browsers have 'session restore' capabilities which will store open tabs/windows between restarts, further complicating 'non-persistent' cookies.

@lesam and I discussed this risk and I advocated adding some additional text to our documentation, which @lesam said he'd take to the documentation team.